lib/transition/utils.c

```diff
@@ -288,3 +290,11 @@ update_abort_priority(crm_graph_t * graph, int priority,
 
     return change;
 }
+
+void
+set_per_node_metadata_cache(GHashTable *cache)
+{
+    CRM_ASSERT(cache != NULL);
+
+    per_node_metadata_cache = cache;
+}
```

Comment:
Elaborating on my idea to register a function here instead, it would be something like:

void crm_register_cache_check_fn(gboolean (*fn)(const char *nodename, lrmd_rsc_info_t *rsc_info))